Mathlib/AlgebraicTopology/ModelCategory/Instances.lean

```diff
@@ -0,0 +1,359 @@
+  exact ⟨fun _ ↦ weakEquivalence_of_precomp f g, fun _ ↦ inferInstance⟩
+
+variable {f g} {fg : X ⟶ Z}
+
+lemma weakEquivalence_of_postcomp_of_fac (fac : f ≫ g = fg)
+    [WeakEquivalence g] [hfg : WeakEquivalence fg] :
+    WeakEquivalence f := by
+  subst fac
+  exact weakEquivalence_of_postcomp f g
+
+lemma weakEquivalence_of_precomp_of_fac (fac : f ≫ g = fg)
+    [WeakEquivalence f] [WeakEquivalence fg] :
+    WeakEquivalence g := by
+  subst fac
+  exact weakEquivalence_of_precomp f g
+
+end HasTwoOutOfThreeProperty
+
+variable [CategoryWithCofibrations C] [CategoryWithFibrations C]
+
+section
+
+variable [IsWeakFactorizationSystem (trivialCofibrations C) (fibrations C)]
+
+lemma fibrations_llp :
+    (fibrations C).llp = trivialCofibrations C := by
+  apply llp_eq_of_wfs
+
+lemma trivialCofibrations_rlp :
+    (trivialCofibrations C).rlp = fibrations C := by
+  apply rlp_eq_of_wfs
+
+instance : (trivialCofibrations C).IsStableUnderCobaseChange := by
+  rw [← fibrations_llp]
+  infer_instance
+
+instance : (fibrations C).IsStableUnderBaseChange := by
+  rw [← trivialCofibrations_rlp]
+  infer_instance
+
+instance : (trivialCofibrations C).IsMultiplicative := by
+  rw [← fibrations_llp]
+  infer_instance
+
+instance : (fibrations C).IsMultiplicative := by
+  rw [← trivialCofibrations_rlp]
+  infer_instance
+
+variable (J : Type w)
+
+instance isStableUnderCoproductsOfShape_trivialCofibrations :
+    (trivialCofibrations C).IsStableUnderCoproductsOfShape J := by
+  rw [← fibrations_llp]
+  apply MorphismProperty.llp_isStableUnderCoproductsOfShape
+
+instance isStableUnderProductsOfShape_fibrations :
+    (fibrations C).IsStableUnderProductsOfShape J := by
+  rw [← trivialCofibrations_rlp]
+  apply MorphismProperty.rlp_isStableUnderProductsOfShape
+
+end
+
+section
+
+variable [IsWeakFactorizationSystem (cofibrations C) (trivialFibrations C)]
+
+lemma trivialFibrations_llp :
+    (trivialFibrations C).llp = cofibrations C := by
+  apply llp_eq_of_wfs
+
+lemma cofibrations_rlp :
+    (cofibrations C).rlp = trivialFibrations C := by
+  apply rlp_eq_of_wfs
+
+instance : (cofibrations C).IsStableUnderCobaseChange := by
+  rw [← trivialFibrations_llp]
+  infer_instance
+
+instance : (trivialFibrations C).IsStableUnderBaseChange := by
+  rw [← cofibrations_rlp]
+  infer_instance
+
+instance : (cofibrations C).IsMultiplicative := by
+  rw [← trivialFibrations_llp]
+  infer_instance
+
+instance : (trivialFibrations C).IsMultiplicative := by
+  rw [← cofibrations_rlp]
+  infer_instance
+
+
+variable (J : Type w)
+
+instance isStableUnderCoproductsOfShape_cofibrations :
+    (cofibrations C).IsStableUnderCoproductsOfShape J := by
+  rw [← trivialFibrations_llp]
+  apply MorphismProperty.llp_isStableUnderCoproductsOfShape
+
+instance isStableUnderProductsOfShape_trivialFibrations :
+    (trivialFibrations C).IsStableUnderProductsOfShape J := by
+  rw [← cofibrations_rlp]
+  apply MorphismProperty.rlp_isStableUnderProductsOfShape
+
+end
+
```

Comment:
Should we also record stability of (trivial) cofibrations under transfinite compositions, using [CategoryTheory.MorphismProperty.isStableUnderTransfiniteCompositionOfShape_llp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/SmallObject/TransfiniteCompositionLifting.html#CategoryTheory.MorphismProperty.isStableUnderTransfiniteCompositionOfShape_llp)?

This might add an import though.